File: 1018.可被-5-整除的二进制前缀.js
/*
 * @lc app=leetcode.cn id=1018 lang=javascript
 *
 * [1018] 可被 5 整除的二进制前缀
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {boolean[]}
 */
var prefixesDivBy5 = function (A) {
  let i = -1;
  let n = 0;
  while (++i < A.length) A[i] = (n = ((n << 1) + A[i]) % 5) === 0;
  return A;
};
// @lc code=end

File: 1018.可被-5-整除的二进制前缀.rs
/*
 * @lc app=leetcode.cn id=1018 lang=rust
 *
 * [1018] 可被 5 整除的二进制前缀
 */

// @lc code=start
impl Solution {
    pub fn prefixes_div_by5(a: Vec<i32>) -> Vec<bool> {
        a.iter().scan(0, |sum, &x| {*sum = (*sum * 2 + x) % 5;Some(*sum == 0)}).collect()
    }
}
// @lc code=end

